source/common/network/connection_impl.cc

```diff
@@ -342,24 +350,24 @@ void ConnectionImpl::readDisable(bool disable) {
     }
   } else {
     --read_disable_count_;
```

Comment:
While you are here can you ASSERT this is greater than 0 before decrementing? I'm surprised this was not already asserted.